src/lib.rs

```diff
@@ -49,34 +61,25 @@ impl Drone for MyDrone {
             packet_recv,
             pdr,
             packet_send,
-            // TODO: I don't understand why we need to use RefCell here???
-            known_flood_ids: RefCell::new(HashSet::new()),
+            known_flood_ids: HashSet::new(),
             state: State::Working,
-            // rng: rand::rng(),
         }
     }
 
     fn run(&mut self) {
         'loop_label: loop {
-            // TODO: use select_biased! to prioritize simulation controller commands
-            select! {
-                recv(self.packet_recv) -> packet_res => {
-                    match packet_res {
-                        Err(_err) => {
-                             match &self.state {
-                                 State::Working => {
-                                     panic!("There is no connected sender to the drone's receiver channel and no DroneCommand::Crash has been received")
-                                 },
-                                 State::Crashing => {
-                                     break 'loop_label
-                                 },
-                             }
-                        }
-                        Ok(packet) => {
-                            self.process_packet(packet);
-                        }
-                    }
-                },
+            /*
+                From https://shadow.github.io/docs/rust/crossbeam/channel/macro.select_biased.html
+                "If multiple operations are ready at the same time, the operation nearest to the front of the list is always selected"
+                So recv(self.controller_recv) needs to come before recv(self.packet_recv)
+
+                WARNING: select_biased! seems to select a channel even if one end of it is dropped
+                         for instance, if I create a drone and drop the Sender<DroneCommand> channel and I keep using the drone,
+                         then the drone will always select the recv(self.controller_recv) arm, since it receives an Err().
+                         This means that the drone stops working properly in the case of bad channels management.
+                         Interestingly, this didn't occur with the select! macro
+             */
+            select_biased! {
                 recv(self.controller_recv) -> command_res => {
                     if let Ok(command) = command_res {
                         match command
```

Comment:
So, the drone selects only the command recv arm, starving the other, just in the case that simulation controller drops the command sender for that drone, right?

Isn't the solution just to add a match for when we get an err here and panic because simulationcontroller dropped the channel or am I missing something?

we could also have a test that verifies that the drone panics when we drop simulation controller command sender

Reply:
Yes, I guess that could work, but there may be the same problem on the other receiver channel. For instance, if the same were to happen within `recv(self.packet_recv)` I think that we would have unexpected panics, so we need to test this a lot. But this issue may also be fixed by matching the type of error returned maybe? We just need to be sure that the receive part behaves as intended hahah

Reply:
We already have that figured out, if we get the error on packet recv we either crash safely if we got the crash command before, otherwise we panic. I believe that this already works as intended, but I agree that we need to write some tests for that. 

Talking about the type of error returned, there is **only one error** with the recv() method, the one that happens when every sender disconnected, and in that case we panick/crash. I don't think this stuff could lead to any unexpected panics